tests/unit/calmBull.test.ts

```diff
@@ -0,0 +1,142 @@
+import { describe, expect, test } from '@jest/globals'
+import { BigNumber, Contract } from 'ethers'
+import { QUOTER } from '../../constants/address'
+import { BIG_ONE, DEFAULT_SLIPPAGE, WETH_DECIMALS_DIFF } from '../../constants/numbers'
+import { Quoter } from '../../types/contracts'
+import quoterAbi from '../../abis/quoter.json'
+import { getLeverageRebalanceDetails } from '../../utils/calmBull'
+import * as quoterFns from '../../utils/quoter'
+import '../../utils/math'
+
+const quoter = new Contract(QUOTER, quoterAbi) as Quoter
+
+describe('CalmBull: Leverage Rebalance', () => {
+  const crabBalance = BIG_ONE.mul(200)
+  const loanCollat = BIG_ONE.mul(200)
+  const loanDebt = BIG_ONE.mul(100_000).div(WETH_DECIMALS_DIFF)
+  const crUpper = BigNumber.from('3000000') // 3
+  const crLower = BigNumber.from('1200000') // 1.2
+  const deltaUpper = BIG_ONE.mul(12).div(10) // 1.2
+  const deltaLower = BIG_ONE.mul(9).div(10) //.9
+
+  function mockQuoterFunctions(ethPrice: BigNumber) {
+    jest
+      .spyOn(quoterFns, 'quoteExactIn')
+      .mockImplementation((quoter: Quoter, tokenIn: string, tokenOut: string, amountIn: BigNumber, poolFee: number) => {
+        return Promise.resolve(amountIn.wdiv(ethPrice))
+      })
+
+    jest
+      .spyOn(quoterFns, 'quoteExactOut')
+      .mockImplementation(
+        (quoter: Quoter, tokenIn: string, tokenOut: string, amountOut: BigNumber, poolFee: number) => {
+          return Promise.resolve(amountOut.wdiv(ethPrice))
+        },
+      )
+  }
+
+  /**
+   * Initial values
+   * ETH price = 1000
+   * crabPrice = 500
+   * sqth price = .1
+   * CR = 2
```

Comment:
maybe good to assert that these are true to ensure test is testing what you think

ie assert that cr = 2 before test and that it goes down when eth goes down, and delta is 1 before, goes up when crab + eth changes, etc

Reply:
Do we have to do that in FE tests? The point of the FE test is to check if the correct rebalance is selected for a given crab price!